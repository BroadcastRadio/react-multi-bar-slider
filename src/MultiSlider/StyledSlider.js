import PropTypes from 'prop-types';
import styled from 'react-emotion';
import transition from '../utils/transition';
import getHalf from '../utils/getHalf';

const StyledSlider = styled('div')({
  position: 'relative',
  height: 14,
  boxSizing: 'border-box',
  transition
}, ({ width, height, backgroundColor, roundedCorners, readOnly, css }) => ({
  width,
  height,
  backgroundColor,
  borderRadius: roundedCorners ? getHalf(height) : 0,
  cursor: readOnly ? 'auto' : 'pointer',
  ...css
}));


StyledSlider.propTypes = {
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  backgroundColor: PropTypes.string.isRequired,
  roundedCorners: PropTypes.bool.isRequired,
  readOnly: PropTypes.bool.isRequired,
  css: PropTypes.object.isRequired
};

StyledSlider.defaultProps = {
  css: null
};

export default StyledSlider;
