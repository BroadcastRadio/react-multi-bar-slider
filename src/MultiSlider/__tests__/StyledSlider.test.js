import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import StyledSlider from '../StyledSlider';

Enzyme.configure({ adapter: new Adapter() });

beforeAll(() => {
  global.styledSliderProps = {
    width: 1000,
    height: 20,
    backgroundColor: 'purple',
    roundedCorners: true,
    readOnly: false,
    css: {}
  };
});

afterAll(() => {
  delete global.styledSliderProps;
});

/* global styledSliderProps */
describe('StyledSlider.js', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<StyledSlider {...styledSliderProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const styledSlider = render(<StyledSlider {...styledSliderProps} />);
    expect(styledSlider).toBeDefined();
  });
});
