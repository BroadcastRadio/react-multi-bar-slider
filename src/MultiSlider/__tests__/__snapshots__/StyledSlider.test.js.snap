// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledSlider.js matches the snapshot 1`] = `
<div
  className="css-p5eloh"
  height={20}
  readOnly={false}
  width={1000}
/>
`;
